src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -45,13 +47,43 @@ internal DurableOrchestrationContext(DurableTaskExtension config, string functio
         }
 
         /// <inheritdoc />
-        public override string InstanceId => this.innerContext.OrchestrationInstance.InstanceId;
+        public override string InstanceId
+        {
+            get
+            {
+                if (this.innerContext != null)
+                {
+                    return this.innerContext.OrchestrationInstance.InstanceId;
+                }
+                else
+                {
+                    return this.instanceId;
+                }
+            }
+
+            internal set => this.instanceId = value;
+        }
 
         /// <inheritdoc />
         public override DateTime CurrentUtcDateTime => this.innerContext.CurrentUtcDateTime;
 
         /// <inheritdoc />
-        public override bool IsReplaying => this.innerContext.IsReplaying;
+        public override bool IsReplaying
+        {
+            get
+            {
+                if (this.innerContext != null)
+                {
+                    return this.innerContext.IsReplaying;
+                }
+                else
+                {
+                    return this.isReplaying;
+                }
+            }
+
+            internal set => this.isReplaying = value;
+        }

```

Comment:
With this change, it's no longer clear what is intended to be the source of truth - the inner context object or the new private members you're adding. I think we have to choose one and go with it or else our future selves (or future contributors) will get confused. There is precedent for mutable properties, such as `ParentOrchestrationId`, so I suggest you follow that model.